packages/gatsby/src/schema/types/built-in-types.js

```diff
@@ -0,0 +1,73 @@
+const fileType = `
+  type File implements Node @infer {
+    sourceInstanceName: String
+    absolutePath: String
```

Comment:
Are all those really nullable? I think many (?) of those might not be. Not sure.

Reply:
That's how `gatsby-plugin-schema-snapshot` printed them. I guess we can change it for the next major but for now, we should probably stay consistent with the current version?

Reply:
Nullable to non-nullable shouldn't be a breaking change if they actually can't be null, I think it's a good idea to have less nullable fields if it's possible.

Reply:
So most stats are an output of `fs.stat`. We should check out TS signature for that and assign non-nullness accordingly.

Reply:
Changed this.

We have our own typings in `gatsby-source-filesystem`: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/index.d.ts#L58-L96

Interestingly `sourceInstanceName` is also non-null. @pieh you mentioned it could be null?